File: problem-5.py
### Problem 5
###-----------------------------------------------------------------------------------------------------------
### 2520 is the smallest number that can be divided by each of the numbers from 1 to 10 without any remainder.
### What is the smallest positive number that is evenly divisible by all of the numbers from 1 to 20?

### Solution

# Check if evenly divisible
def isDivisible(n):
	for i in range(1, 21):
		if n % i != 0:
			return False

	return True

# Calculate smallest divisible integer
number = 20

while True:
	if isDivisible(number):
		break
	else:
		number += 20

print(number)

File: java/problem-5.java
public class Problem5{

	public static boolean isDivisible(int n) {

		if (n%3 == 0 && n%6 == 0 && n%7 == 0 && n%8 == 0 && n%9 == 0) {
			if (n%11 == 0 && n%12 == 0 && n%13 == 0 && n%14 == 0 && n%15 == 0) {
				if (n%16 == 0 && n%17 == 0 && n%18 == 0 && n%19 == 0) {
					return true;
				}
			}
		}
		return false;

	}

	public static void main(String[] args) {

		int n = 20;

		while (true) {
			if (isDivisible(n)) {
				break;
			}

			n += 20;

		}

		System.out.println(n);

	}
	

}